File: golang/vaas/pkg/messages/verdict_request_for_url.go
// Package messages provides structures for handling communication messages between the client and the VaaS server.
package messages

import (
	"github.com/GDATASoftwareAG/vaas-go/pkg/options"
	"github.com/google/uuid"
)

// VerdictRequestForURL is a specific implementation of VerdictRequest used for URL analysis requests.
type verdictRequestForURL struct {
	Kind                     Kind                     `json:"kind" default:"VerdictRequestForUrl"`
	URL                      string                   `json:"url"`
	GUID                     string                   `json:"guid"`
	SessionID                string                   `json:"session_id"`
	VerdictRequestAttributes VerdictRequestAttributes `json:"verdict_request_attributes"`
	UseCache                 bool                     `json:"use_cache"`
	UseHashLookup            bool                     `json:"use_shed"`
}

// GetGUID returns the GUID of the verdictRequestForURL.
func (r verdictRequestForURL) GetGUID() string {
	return r.GUID
}

// NewVerdictRequestForURL creates a new verdictRequestForURL instance.
func NewVerdictRequestForURL(sessionID string, options options.VaasOptions, URL string) VerdictRequest {
	return verdictRequestForURL{
		Kind:          VerdictRequestForURLKind,
		URL:           URL,
		SessionID:     sessionID,
		GUID:          uuid.New().String(),
		UseCache:      options.UseCache,
		UseHashLookup: options.UseHashLookup,
	}
}

// NewVerdictRequestForURLWithAttributes creates a new verdictRequestForURL instance with attributes.
func NewVerdictRequestForURLWithAttributes(sessionID string, options options.VaasOptions, URL string, attributes VerdictRequestAttributes) VerdictRequest {
	return verdictRequestForURL{
		Kind:                     VerdictRequestForURLKind,
		URL:                      URL,
		SessionID:                sessionID,
		GUID:                     uuid.New().String(),
		UseCache:                 options.UseCache,
		UseHashLookup:            options.UseHashLookup,
		VerdictRequestAttributes: attributes,
	}
}

File: java/src/main/java/de/gdata/vaas/messages/VerdictRequestForUrl.java
package de.gdata.vaas.messages;

import com.google.gson.Gson;
import com.google.gson.annotations.SerializedName;
import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;

import java.net.URL;
import java.util.UUID;

@Getter
public class VerdictRequestForUrl extends MessageType {
    @NonNull
    String url;
    @Setter
    @NonNull
    @SerializedName("session_id")
    String sessionId;
    @NonNull
    String guid;
    @Setter
    @SerializedName("verdict_request_attributes")
    VerdictRequestAttributes verdictRequestAttributes;
    @SerializedName("use_hash_lookup")
    transient boolean UseHashLookup;
    @SerializedName("use_cache")
    boolean UseCache;

    public VerdictRequestForUrl(URL url, String sessionId, UUID guid) {
        super(Kind.VerdictRequestForUrl);
        this.sessionId = sessionId;
        this.guid = guid.toString();
        this.url = url.toString();
    }

    public VerdictRequestForUrl(URL url, String sessionId, UUID guid,
            VerdictRequestAttributes verdictRequestAttributes) {
        this(url, sessionId, guid);
        this.verdictRequestAttributes = verdictRequestAttributes;
    }

    public VerdictRequestForUrl(URL url, String sessionId, UUID guid, VaasOptions options) {
        this(url, sessionId, guid);
        this.UseCache = options.UseCache;
        this.UseHashLookup = options.UseHashLookup;
    }

    public VerdictRequestForUrl(URL url, String sessionId, UUID guid, VerdictRequestAttributes verdictRequestAttributes,
                                VaasOptions options) {
        this(url, sessionId, guid);
        this.verdictRequestAttributes = verdictRequestAttributes;
    }

    public String toJson() {
        return new Gson().toJson(this);
    }
}